corehq/apps/userreports/management/commands/delete_orphaned_ucrs.py

```diff
@@ -32,58 +32,82 @@ def add_arguments(self, parser):
             help='Only check this DB engine',
         )
         parser.add_argument(
-            '--drop-empty-tables',
+            '--force-delete',
             action='store_true',
             default=False,
-            help='Call DROP TABLE on tables with no rows',
+            help='Drop orphaned tables on active domains'
         )
         parser.add_argument(
-            '--drop-deleted-tables',
+            '--dry-run',
             action='store_true',
             default=False,
-            help='Call DROP TABLE on tables from deleted domains',
+            help='Do not modify the DB if set to true',
         )
 
     def handle(self, **options):
         data_sources = list(DataSourceConfiguration.all())
         data_sources.extend(list(StaticDataSourceConfiguration.all()))
-        tables_by_engine_id = get_tables_by_engine_id(data_sources, options.get('engine_id'))
-        tables_to_remove_by_engine = get_tables_to_remove_by_engine(tables_by_engine_id)
-        prune_tables(tables_to_remove_by_engine, options['drop_empty_tables'], options['drop_deleted_tables'])
+        tables_by_engine_id = get_tables_for_data_sources(data_sources, options.get('engine_id'))
+        orphaned_tables_by_engine_id = get_tables_without_data_sources(tables_by_engine_id)
+        drop_tables(orphaned_tables_by_engine_id, force_delete=options['force_delete'], dry_run=options['dry_run'])
 
 
-def prune_tables(tables_to_remove_by_engine, drop_empty_tables, drop_deleted_tables):
-    deleted_domains = Domain.get_deleted_domain_names()
+def drop_tables(tables_to_remove_by_engine, force_delete=False, dry_run=False):
+    """
+    :param tables_to_remove_by_engine: {'<engine_id>': [<tablename>, ...]}
+    :param force_delete: if True, delete orphaned tables for active domains
+    :param dry_run: if True, do not make changes to the DB
+    """
+    dry_run_tag = '[DRY_RUN]' if dry_run else ''
+
     for engine_id, tablenames in tables_to_remove_by_engine.items():
-        print("\nTables no longer referenced in database: {}:\n".format(engine_id))
+        print(f"Looking at engine {engine_id}")
```

Comment:
nit: i don't know that people know what an "engine", the old wording of database might be clearer